### ISSUE TITLE
Sensors recycleapp no longer loading after updates of 6 December 2024

### ISSUE THREAD

**Issue by XalaTheShepard** [pippyn/Home-Assistant-Sensor-Afvalbeheer#536]
Dear all, 

After the latest updates, the sensors of Recycleapp are no longer loaded. 
They worked fine up untill version 5.5.1 of the integration. 
I tried to revert back to version 5.5.1, but HACS won't allow the action. 
In the logs I find no errors regarding Afvalbeheer. 

Thanks for looking into the issue. 

Kind regards, 

Robin

**Comment by Zixim** [comment]
on 5.5.3 Recycleapp is working, for me.

**Comment by pippyn** [comment]
- remove all sensors created by this integration.
- update the sensor to v5.5.4
- restart HA

**Comment by pippyn** [comment]
v5.5.1 has major issues and shouldn't be used!
What can try the following
- remove the sensor config from your yaml file
- remove all sensors created by this integration (important!)
- delete the integration
- restart HA
- install v5.5.4 of the sensor
- restart HA
- add your config again
- restart HA

**Comment by Zixim** [comment]
I did this (i'm not using HACS) : 

- delete afvalbeheer folder from custom_components folder.
- restart Home Assistant
- delete the afvalbeheer/recycleapp entities from the HA entities list.
- copy the Afvalbeheer folder (5.5.4) into custom_components
- Restart Home Assistant
All good.

**Comment by XalaTheShepard** [comment]
Still the same issue. I have now version 5.5.4 of the integration, but no Recycleapp entities nor sensors. Followed everything step by step, exept for deleting the sensors / entities (how can I delete non-existing sensors / entities)? Still no sensors showing up.

**Comment by Zixim** [comment]
After you delete the integration & restart, you can then delete the entities/sensors from the HA webif.

**Comment by pippyn** [comment]
You can find them here:
<a href="https://my.home-assistant.io/redirect/entities/" target="_blank" rel="noreferrer noopener"><img src="https://my.home-assistant.io/badges/entities.svg" alt="Open your Home Assistant instance and show your entities." /></a>

**Comment by XalaTheShepard** [comment]
I did check the entities and sensors in that menu, but they aren't there. Not with the integration, not without.

**Comment by Zixim** [comment]
is your .yaml config still in place ?

**Comment by XalaTheShepard** [comment]
Yes. Just deleted for the second time the config of Afvalbeheer.

**Comment by Zixim** [comment]
I meant to say, if you deleted .yaml then do not forget to recreate it and restart.
But in fact, there is no reason to delete the .yaml.

**Comment by XalaTheShepard** [comment]
First time I marked them with "#". Now I try it with deleting it and setting it up from scratch again.

**Comment by XalaTheShepard** [comment]
And every time I save it after the adjustments and perform a reboot.

**Comment by Zixim** [comment]
for info : 
```:yaml
afvalbeheer:
    wastecollector: RecycleApp       #(required)
    printwastetypes: 0               #sensor will print a list of all possible waste fractions for your address and waste collector on every restart
    resources:                       #(at least 1 required)
      - restafval
      - gft
      - papier
      - pmd
      - grofvuil
      - takken
      - textiel
    postcode: redacted                   #(required)
    streetname: redacted       #required for Ophaalkalender.be and Limburg.NET
    streetnumber: redacted                 #(required)
    # suffix: ''                       #(optional)
    upcomingsensor: 1                #(optional)
    dateformat: '%d-%m-%Y'           #(optional)
    dateonly: 0                      #(optional)
    dayofweek: 1
    dateobject: 0                    #(optional)
    name: ''                         #(optional)
    nameprefix: 1                    #(optional)
    builtinicons: 1                  #(optional)
    disableicons: 0                  #(optional)
    dutch: 0                         #(optional)
    updateinterval: 12
```

**Comment by Zixim** [comment]
_if_ 5.5.1 ends up working, then there's no valid reason why 5.5.4 wouldn't also work.

**Comment by XalaTheShepard** [comment]
If that is the case, why do I get the sensors with version 5.5.1 and not with the later versions?

**Comment by Zixim** [comment]
That's what troubleshooting is for :shrug: 
I'm not saying "don't try 5.5.1", I'm just saying that 5.5.1 is broken so you can't stay on that version.

**Comment by XalaTheShepard** [comment]
I am fully aware, and really want version 5.5.4 to be also working for me. But at this moment, I have tried everything and still no luck with version 5.5.4. All my entities just go missing after version 5.5.1 and I can not explain why.

**Comment by XalaTheShepard** [comment]
![Probleem afvalbeheer 554](https://github.com/user-attachments/assets/f7bc37c1-081b-40aa-9c25-3bd44fed10d7)

**Comment by Zixim** [comment]
do the entities contain data (see dev tools) ?

**Comment by XalaTheShepard** [comment]
As compared to the image of version 5.5.4, the entities pop up in version 5.5.1 , but no longer in the later versions. I wish I knew what is causing this issue.

**Comment by pippyn** [comment]
Any errors in your logs?

**Comment by XalaTheShepard** [comment]
The entities of version 5.5.1 contain data in the developer tool (date, icon). But no days untill.

**Comment by XalaTheShepard** [comment]
No errors in the logs with version 5.5.1

**Comment by XalaTheShepard** [comment]
I will try updating to version 5.5.4 once again and provide you the logs.

**Comment by XalaTheShepard** [comment]
Updated back to 5.5.4 again. Entities are missing again and no errors in the logs regarding Afvalbeheer.

**Comment by Zixim** [comment]
And you made sure that you manually deleted the entities before installing 5.5.4 ?

I donno... surely @pippyn will chime in.

**Comment by XalaTheShepard** [comment]
I followed every step, but could only delete half of the entities. The other half I was unable to delete.

**Comment by Zixim** [comment]
"The other half I was unable to delete."
That isn't ok. They should be deleted.

**Comment by pippyn** [comment]
Deleting the remaining entities is the most imported step.
Why can't you delete the other ones?

**Comment by XalaTheShepard** [comment]
Because I get an error it was not set up using the UI

**Comment by Zixim** [comment]
delete the integration, restart, delete _all_ entities.

**Comment by XalaTheShepard** [comment]
After deleting the integration, all entities are gone as well.

**Comment by Zixim** [comment]
That's strange, because the entities can't auto-delete themselves.
It's a manual thingy.

**Comment by XalaTheShepard** [comment]
Trying again at the moment. Just deleted the integration and HA is rebooting.

**Comment by XalaTheShepard** [comment]
![Probleem afvalbeheer 554 2](https://github.com/user-attachments/assets/0974236f-bf2a-4edb-b606-18b28e6f5955)

**Comment by Zixim** [comment]
what does the entities list look like ?
![image](https://github.com/user-attachments/assets/a598c60e-cf68-4584-be20-f0e281f27c4f)

**Comment by XalaTheShepard** [comment]
I just deleted the config of Afvalbeheer and am rebooting. Will have a look after reboot.

**Comment by Zixim** [comment]
don't delete your yaml config.

**Comment by XalaTheShepard** [comment]
OK, will start over again.

**Comment by Zixim** [comment]
This is what it should look like, after deleting the integration & rebooting.
![image](https://github.com/user-attachments/assets/643a3131-1d0c-480c-98f1-7582bdd10b98)

Then you can select them all, and delete them.
Restart, reinstall the integration, restart.done.

**Comment by XalaTheShepard** [comment]
OK, just reinstalled version 5.5.1 . Will now delete it in HACS -> reboot -> and try to delete the entities (will provide screenshot)

**Comment by XalaTheShepard** [comment]
I had to re-enable the remaining entities and reboot HA. Hopefully then I will be able to delete them.

**Comment by XalaTheShepard** [comment]
Now, integration is deleted (config still in YAML) and all integrations are deleted. Will reboot again and reïnstall the Afvalbeheer integration.

**Comment by XalaTheShepard** [comment]
So far, so good. Now I have my entities back. Many thanks for the support, because I never could have figured that one out myself :)
